libgnucash/backend/xml/sixtp-dom-parsers.cpp

```diff
@@ -92,6 +98,10 @@ dom_tree_to_guid (xmlNodePtr node)
 static KvpValue*
 dom_tree_to_integer_kvp_value (xmlNodePtr node)
 {
+    if (auto txt = dom_node_to_text (node))
+        if (gint64 daint; string_to_gint64 (txt, &daint))
+            return new KvpValue{daint};
+
```

Comment:
This is sloppy. At least put the new code after the decls and use `text` instead of creating a new `txt` and `daint`. Better yet would be 
```
 {
    gchar* text{dom_node_to_text (node)};
    gint64 daint;
    KvpValue* ret = nullptr;
    gboolean valid_int{FALSE};
    if (text)
        valid_int = string_to_gint64 (text, &daint);
    else
    {
        text = dom_tree_to_text (node);
        valid_int = string_to_gint64 (text, &daint);
        g_free (text);
    }
    if (valid_int)
        ret = new KvpValue {daint};
    return ret;
}
```
because it preserves the RV optimization achieved by pre-declaring ret.

Reply:
> hmm then text will be a const char* or std::string?

Oh, right. It would have to be 
```
     std::string tree_text{};
     std::string_view text{dom_node_to_text(node)};
     if (text.empty())
     {
          tree_text = dom_tree_to_text(node);
          text = tree_text;
      }
      ...
```

Reply:
> > hmm then text will be a const char* or std::string?
> 
> Oh, right. It would have to be
> 
> ```
>      std::string tree_text{};
>      std::string_view text{dom_node_to_text(node)};
>      if (text.empty())
>      {
>           tree_text = dom_tree_to_text(node);
>           text = tree_text;
>       }
>       ...
> ```

Not quite because the dom_node_to_text(node) needs to be freed